ytt/001-schemas/README.md

```diff
@@ -0,0 +1,1647 @@
+- respect schema for data values set via cmd line flags/env vars
+- add --data-values-schema-inspect (similar to --data-values-inspect)
+  - generate html view via builtin server
+- with the addition of a schema, the first data values file no longer serves that role. Could we improve the UX of adding multiple data value files by not requiring subsequent data values to be overlays?
+- could a user include both a data/value + schema/amend in the same file?
+  - allowing a user to define and include input values in one file?
+
+
+
+# Sample Usage
+
+Vetting of User eXperience by way of examples.
+
+## Common Use Cases
+
+Expectedly common situations that invoke more than one feature, illustrating proper interactions.
+
+### Mutually Exclusive Map Configuration
+
+## Edge Cases
+
+Situations that stretch the capabilities of the feature set.
+
+### Prometheus Label Set Rewriting
+
+Prometheus includes a feature allowing operators to articulate label rewritting rules for metrics.
+This feature is used throughout Prometheus' configuration. It is centered around a structure
+known as a "relabeling".
+
+Defining edits that operate on runtime text is frought with errors.
+- a minor typo of a keyword,
+- a regular expression that fails only in certain cases, or
+- a sometimes required configuration being omitted
+
+This example illustrates how Schema would enhance the Prometheus User's configuration experience.
+
+Sources:
+- https://github.com/prometheus-community/helm-charts/blob/451448b53/charts/kube-prometheus-stack/values.yaml
+- https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
+
+```yaml=
+#! ==- Relabel Config -==
+#!
+
+#@ def replace_has_target_label(relabel):
+#@   if relabel.action == "replace" and "targetLabel" not in relabel:
+#@     return (False, "'targetLabel' not found on replace relabel rule")
+#@   end
+#@ end
+
+#@ def non_hashmod_has_regex(relabel):
+#@   if relabel.action in ["replace", "keep", "drop", "labelmap", "labeldrop", "labelkeep"] and \
+#@      "regex" not in relabel:
+#@     return (False, "'regex' not found on {} relabel rule".format(relabel.action))
+#@   end
+#@ end
+
+#! As of Prometheus 2.22, there are 200+ meta labels
+#@ known_meta_labels = {
+#@   "__meta_azure_machine_id": True,
+#@   "__meta_azure_machine_location": True,
+#@   "__meta_azure_machine_name": True,
+#@   "__meta_azure_machine_os_type": True,
+#@   "__meta_azure_machine_private_ip": True,
+#!   ... and many more
+#@ }
+#@ def known_meta_label(label):
+#@   if label.startswith("__meta_"):
+#@     if known_meta_labels[label] == None:
+#@       return (False,
+#@               "{} is not a known meta label (did you mean {}?)"\
+#@                 .format(label, spelling.nearest(label, known_meta_labels.keys())))
+#@     end
+#@   end
+#@ end
+
+---
+#@ def relabelings():
+#@schema/validate replace_has_target_label, non_hashmod_has_regex
+- 
+  #@schema/title "Source labels"
```

Comment:
Break this out into two functions: one for the contained structure and another for the collection.